feat: 데이터 로드 후 compound document 핸들링

cmygray opened the pull request:
changes:
- 리스트, 단일 데이터에 대해 included 필드를 store module에 저장
- included 필드와 해당 모듈이 모두 존재할 때만 동작

references:
- https://github.com/reststate/reststate-vuex/pull/38

Signed-off-by: cmygray <cmygray@gmail.com>

cmygray commented:
TODOS:
- [ ] STORE_RECORD mutation payload에 included를 포함
- [ ] STORE_RECORD mutation plugin을 통해 included 관리

dependabot-support committed:
Update @reststate/client requirement from ^0.0.6 to ^0.0.7

Updates the requirements on [@reststate/client](https://github.com/reststate/reststate-client) to permit the latest version.
- [Release notes](https://github.com/reststate/reststate-client/releases)
- [Commits](https://github.com/reststate/reststate-client/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
Merge pull request #40 from reststate/dependabot/npm_and_yarn/@reststate/client-tw-0.0.7

Update @reststate/client requirement from ^0.0.6 to ^0.0.7

CodingItWrong committed:
Fix SyntaxError on import

CodingItWrong committed:
Bump client

Commit:
Merge pull request #43 from reststate/bump-client-3

Bump client 3

CodingItWrong committed:
Bump version to 0.0.14

CodingItWrong committed:
Bump version to 0.0.15

cmygray committed:
feat: handle compound document after loadById, loadRelated

changes:
- add new action 'storeIncluded'
- add pluralize.js to get resource name from resource type

Signed-off-by: cmygray <cmygray@gmail.com>

cmygray committed:
refactor: extract list data handler

changes:
- add new action 'handleCompoundDocument'

Signed-off-by: cmygray <cmygray@gmail.com>

CodingItWrong committed:
Update cached page when options change

Commit:
Merge pull request #44 from reststate/page-options

Fix order of records in `page` getter

CodingItWrong committed:
Bump version to 0.0.16

CodingItWrong committed:
Update example versions

CodingItWrong committed:
Update example code to show widgets

junheon commented on src/reststate-vuex.js:
```diff
@@ -253,6 +257,102 @@ const resourceModule = ({ name: resourceName, httpClient }) => {
+
+      storeIncluded({ commit, dispatch }, { data, included }) {
+        Object.keys(data.relationships).forEach(key => {
+          if (!data.relationships[key].data) delete data.relationships[key];
```
call by reference이기 때문에 relationships에 존재하지만 included에 존재하지 않는경우 원본 relationships를 지우는 문제가 발생합니다.

원본 relationships를 삭제하면 related와 compound document를 사용할 경우 문제가 생기겠지요.
ex) collection의 vendor.cover_image는 compound document로 products는 related document로 처리할경우 products의 relationships가 삭제됨

새로운 map을 사용하는 대신 object에 대한 이터레이션은 다음과 같이 처리할 수 있습니다.
```
Object.keys(data.relationships).forEach(key => {
  /* 또는 for in loop 로 성능은 for in loop 가 더 빠름  */
  const relationship = data.relationships[key];

  if (!relationship.data) { return; }

  const { data } = relationship;

  if (Array.isArray(data) {
    // logic
  } else {
    // logic
  }
}
```

cmygray committed:
fix: preserve relationship even if it doesn't contain any data

Signed-off-by: cmygray <cmygray@gmail.com>

cmygray committed:
feat: store included record using array iterator

changes:
- iterate included data(array) to record data
- iterate relationships data(object) to record related data

Signed-off-by: cmygray <cmygray@gmail.com>

cmygray committed:
feat: related 저장 시 일관된 parent 구조 사용

changes:
- (loadRelated) record id, type 만을 parent 로 사용

Signed-off-by: cmygray <cmygray@gmail.com>

cmygray committed:
feat: compound document 처리 시 related 데이터 중복 제거

changes:
- (storeIncluded) included 이터레이션에서는 related를 저장하지 않음

Signed-off-by: cmygray <cmygray@gmail.com>

dependabot-support committed:
Update prettier requirement from 1.17.0 to 1.17.1

Updates the requirements on [prettier](https://github.com/prettier/prettier) to permit the latest version.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/1.17.0...1.17.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

CodingItWrong committed:
Merge pull request #46 from reststate/dependabot/npm_and_yarn/prettier-1.17.1

Update prettier requirement from 1.17.0 to 1.17.1
